src/lib.rs

```diff
@@ -246,6 +247,14 @@ impl<T: PartialEq> PartialEq for Cactus<T> {
 
 impl<T: Eq> Eq for Cactus<T> {}
 
+impl<T: Hash> Hash for Cactus<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        for v in self.vals() {
+            Hash::hash(v, state);
```

Comment:
why not `v.hash(state)` like in the rust docs?

Reply:
Um, because I copy and pasted that bit from somewhere in the standard library ;) You're right, `v.hash(state)` is clearer (and it works!).